core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -392,19 +392,20 @@ class MapUnit {
             tile.improvementInProgress!!.startsWith("Remove") -> {
                 val tileImprovement = tile.getTileImprovement()
                 if (tileImprovement != null
-                        && tileImprovement.terrainsCanBeBuiltOn.contains(tile.terrainFeature)
+                        && tile.terrainFeatures.any { tileImprovement.terrainsCanBeBuiltOn.contains(it) }
                         && !tileImprovement.terrainsCanBeBuiltOn.contains(tile.baseTerrain)) {
                     tile.improvement = null // We removed a terrain (e.g. Forest) and the improvement (e.g. Lumber mill) requires it!
                     if (tile.resource != null) civInfo.updateDetailedCivResources()        // unlikely, but maybe a mod makes a resource improvement dependent on a terrain feature
                 }
                 if (tile.improvementInProgress == "Remove Road" || tile.improvementInProgress == "Remove Railroad")
                     tile.roadStatus = RoadStatus.None
                 else {
-                    // We put "tile.terrainFeature!=null" because of a strange edge case that SHOULD be solved from 3.11.11+, so we should remove it then and see
-                    if (tile.terrainFeature != null && tile.tileMap.gameInfo.ruleSet.terrains[tile.terrainFeature!!]!!.uniques
-                                    .contains("Provides a one-time Production bonus to the closest city when cut down"))
-                        tryProvideProductionToClosestCity()
-                    tile.terrainFeature = null
+                    val removedFeature = tile.terrainFeatures.find { tile.tileMap.gameInfo.ruleSet.terrains[it]!!.uniques
```

Comment:
I think there's a bit of a mix up
The removed feature should be the one specified in the name of the improvement, tile.improvementInProgress.replace("Remove ","") or something (removePrefix? I remember there's something like that but not sure in what language)
And then, we should check tryProvideProductionToClosestCity on that.

In the current code, if I have features A and B, and I have "Remove A" but B happens to have the unique, B will be removed. Also if I'm reading this correctly remove jungle marsh and fallout won't work?

Reply:
Yea you are right
done